backend/cloudinary/cloudinary.go

```diff
@@ -237,20 +364,25 @@ func (f *Fs) List(ctx context.Context, dir string) (fs.DirEntries, error) {
 }
 
 // getCLDAsset finds the asset at Cloudinary. If it can't be found it returns the error fs.ErrorObjectNotFound.
-func (f *Fs) getCLDAsset(ctx context.Context, remote string) (*admin.SearchAsset, error) {
+func (f *Fs) getCLDAsset(ctx context.Context, remote string, retry int8) (*admin.SearchAsset, error) {
 	// Use the Search API to get the specific asset by display name and asset folder
+
 	searchParams := search.Query{
 		Expression: fmt.Sprintf("asset_folder=\"%s\" AND display_name=\"%s\"",
-			strings.TrimLeft(path.Join(f.root, path.Dir(remote)), "/"),
-			path.Base(remote)),
+			f.FromStandardFullPath(cldPathDir(remote)),
+			CloudinaryEncoder.FromStandardName(f, path.Base(remote))),
 		MaxResults: 1,
 	}
 	results, err := f.cld.Admin.Search(ctx, searchParams)
+	if f.opt.OptimisticSearch && len(results.Assets) == 0 && retry < 3 {
+		time.Sleep(1 * time.Second)
+		return f.getCLDAsset(ctx, remote, retry+1)
+	}
 	if err != nil || len(results.Assets) == 0 {
 		return nil, fs.ErrorObjectNotFound
 	}
 
-	if len(results.Assets) > 1 {
+	if results.NextCursor != "" {
```

Comment:
Do we want to remove this?

Reply:
removing

Reply:
This is mandatory, similar to the (*Fs) String. I revised the implementation